### DISCUSSION TITLE
Removal of unnecessary features for ASUS TUF A16 Advantage Edition

### DISCUSSION THREAD

**Discussion by mikolaj612** [seerge/g-helper#2922] [Q&A]
Hello.
Perhaps not necessarily a bug in the application itself as probably a simple oversight, the problem itself applies only to Advantage Edition models, which appear as "FA617" models.

The first issue is the keyboard - here there is only a single zone, backlit in white, which supports only three effects:
- static,
- breathing,
- strobe.
There is no way to change the color, there is only a 4-step brightness adjustment.
It's hard for me to say whether the animation speed function is responsible for anything with this keyboard.

In addition, the keyboard does not support an on/off effect, there is only an effect when the computer is asleep.
The laptop also does not have a sound effect when it starts up (boot-up sound), and there is no Overdrive option for the screen - at least for the version with a 165Hz screen, which is the most popular, I don't know how it looks like for the variant with a 240Hz screen - but there is no mention for that feature on ASUS webiste.
165Hz panel is the same panel ASUS used for cheaper version of ROG Strix G16.

The other features work well and are aptly sized for this model.

Have a nice day and thank you for your effort, your app is simply great!

**Comment by seerge** [comment]
@mikolaj612 hello, do you know if all A16 Advantage Edition are like that or some have RGB backlight after all ? Cause if so - it's better to keep color selection option, as it doesn't harm anyone :)

**Reply by mikolaj612** [reply]
Every TUF A16 Advantage Edition uses white backlit keyboard ;)
At least that ASUS website states on global webpage and my local variant.

Same goes for Overdrive and other listed features.

**Reply by seerge** [reply]
@mikolaj612 ok, this build should show only supported backlight methods

[GHelper.zip](https://github.com/user-attachments/files/16515737/GHelper.zip)

**Reply by mikolaj612** [reply]
Alright - only supported modes are now available.
However there is still overdrive present and animations that are not supported (keyboard is pulsing while laptop is sleeping - hard to tell if I should count it as a animation).

Also boot sound is not supported on this model.

**Reply by seerge** [reply]
@mikolaj612 app shows Overdrive option based if it's reported by BIOS or not. If it shows it - apparently it's present in BIOS

Pulsing during sleep - is a standard sleep animation for all Asus laptops. You can turn it off under `Extra`, but it can't be customized.

**Reply by mikolaj612** [reply]
> @mikolaj612 app shows Overdrive option based if it's reported by BIOS or not. If it shows it - apparently it's present.
> 
> Pulsing during sleep - is a standard sleep animation for all Asus laptops. You can turn it off under `Extra`, but it can't be customized.

Good to know - ASUS never mentioned it on their spec, also Armoury Crate is not showing it anywhere, in the past I had ROG G15 from 2021 and I had overdrive button available.

So there is no need for all four animation settings when there is only one available?
When power on/off keyboard have only static, lit light.

**Reply by seerge** [reply]
@mikolaj612 

It should be static / breathing / and pulsing animations available for your device. 

As for `Extra` settings - shutdown / startup usually does a single or a double blink when you turn device ON and OFF. All ASUS models support those checkboxes, there is no need to remove them

**Reply by mikolaj612** [reply]
> @mikolaj612
> 
> It should be static / breathing / and pulsing animations available for your device.
> 
> As for `Extra` settings - shutdown / startup usually does a single or a double blink when you turn device ON and OFF. All ASUS models support those checkboxes, there is no need to remove them

Interesting - but I tried and when powering there is no blink - keyboard is just lit at max brightness.
Same goes when powering off it will just go down without any fancy effects.

TUF A16 have very, very basic keyboard it remind me old Legion 5 keyboard.
4 levels of brightness is all it have.
I also tried to enable boot sound but it's not supported.

How may I check overdrive to be working? Armoury Crate is not providing any option to enable it, but if it's working it will be a beneficial upgrade for this mid tier IPS panel.

**Reply by seerge** [reply]
@mikolaj612 it may be permanently be turned ON. Given that display has 3ms response time :)

**Reply by mikolaj612** [reply]
> @mikolaj612 it may be permanently be turned ON. Given that display has 3ms response time :)

My unit came with 165Hz 7ms panel - same is used with non-Nebula G16.
But you might be right - it could be enabled by default and switch is just hidden from plain sight.

**Reply by mikolaj612** [reply]
@seerge you can also safely remove boot sound option from extra tab - not supported anyway (no option to enable in BIOS or AC).
Even when enabled trough your app there will be no boot sound like in ROG machines.
I think it's also safe to remove animations - I tried to (re)enable them and keyboard do not act like it have any effect while power on/off, not even a superfast blink etc.
So only awake and sleeping states are working, but no other customization is available.
Only while sleeping keyboard will pulse from time to time, playing with awake mode will simple enable/disable backlight.

Other options are working great as far as I can tell - great work, been using your app for more than a year, so thank you veeery much ;)

PS: I know that this might not be a priority, but this Advantage Laptop is using two MUX modes, one is (MS)Hybrid and second is SmartAccessGraphic - in theory they switch one to another when playing with GPU modes.
Hybrid is mostly used to save battery life, while AMD SAG is used to provide better performance, because it tries to detected whenever to use dGPU or iGPU.

That switch is available trough ASUS AC, would be nice to have it here, otherwise you need to go/check AMD panel everytime.
